authority/admin/api/admin.go

```diff
@@ -0,0 +1,139 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/go-chi/chi"
+	"github.com/smallstep/certificates/api"
+	"github.com/smallstep/certificates/authority/admin"
+	"go.step.sm/linkedca"
+)
+
+// CreateAdminRequest represents the body for a CreateAdmin request.
+type CreateAdminRequest struct {
+	Subject     string              `json:"subject"`
+	Provisioner string              `json:"provisioner"`
+	Type        linkedca.Admin_Type `json:"type"`
+}
+
+// Validate validates a new-admin request body.
+func (car *CreateAdminRequest) Validate() error {
+	return nil
+}
+
+// GetAdminsResponse for returning a list of admins.
+type GetAdminsResponse struct {
+	Admins     []*linkedca.Admin `json:"admins"`
+	NextCursor string            `json:"nextCursor"`
+}
+
+// UpdateAdminRequest represents the body for a UpdateAdmin request.
+type UpdateAdminRequest struct {
+	Type linkedca.Admin_Type `json:"type"`
+}
+
+// Validate validates a new-admin request body.
+func (uar *UpdateAdminRequest) Validate() error {
+	return nil
+}
+
+// DeleteResponse is the resource for successful DELETE responses.
+type DeleteResponse struct {
+	Status string `json:"status"`
+}
+
+// GetAdmin returns the requested admin, or an error.
+func (h *Handler) GetAdmin(w http.ResponseWriter, r *http.Request) {
+	id := chi.URLParam(r, "id")
+
+	adm, ok := h.auth.LoadAdminByID(id)
+	if !ok {
+		api.WriteError(w, admin.NewError(admin.ErrorNotFoundType,
+			"admin %s not found", id))
+		return
+	}
+	api.JSON(w, adm)
+}
+
+// GetAdmins returns a segment of admins associated with the authority.
+func (h *Handler) GetAdmins(w http.ResponseWriter, r *http.Request) {
+	cursor, limit, err := api.ParseCursor(r)
+	if err != nil {
+		api.WriteError(w, admin.WrapError(admin.ErrorBadRequestType, err,
+			"error parsing cursor and limit from query params"))
+		return
+	}
+
+	admins, nextCursor, err := h.auth.GetAdmins(cursor, limit)
+	if err != nil {
+		api.WriteError(w, admin.WrapErrorISE(err, "error retrieving paginated admins"))
+		return
+	}
+	api.JSON(w, &GetAdminsResponse{
```

Comment:
Protojson?

Reply:
Can't use it. Throws an error.

Reply:
Makes sense, a solution to this, to get a nicer JSON is to use a proto message like the [AdminList](
https://github.com/smallstep/linkedca/blob/27424aae8d2588aad217c75897cba5eebcb9aa91/admin.proto#L24-L26). With a new cursor property.